sdk/search/azure-search/src/main/java/com/azure/search/implementation/SearchApiKeyPipelinePolicy.java

```diff
@@ -1,12 +1,13 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.search;
+package com.azure.search.implementation;
```

Comment:
We shouldn't move this into implementation as a user may want to use this to create their own custom pipeline. Also, this is going to trigger a CheckStyle rule to fail that validates all `HttpPipelinePolicy` implementations are in the public API.

Reply:
We made this class final. How do customer make use of the policy?

Reply:
They may want to create their own pipeline and code such as this:

```java
HttpPipeline myCustomSearchPipeline = new HttpPipelineBuilder()
    .addPolicy(new SearchApiKeyPipelinePolicy(myCredentials)
    .addPolicy(myOtherPolicies)
    .build()
```

This is completely allowed and in some cases expected for a customer to roll their own custom pipeline to handle specific cases they want to cover, remove policies our builders add that they don't want, etc.